lib/events.js

```diff
@@ -497,29 +497,49 @@ function unwrapListeners(arr) {
 
 function once(emitter, name) {
   return new Promise((resolve, reject) => {
-    const eventListener = (...args) => {
-      if (errorListener !== undefined) {
-        emitter.removeListener('error', errorListener);
-      }
-      resolve(args);
-    };
-    let errorListener;
-
-    // Adding an error listener is not optional because
-    // if an error is thrown on an event emitter we cannot
-    // guarantee that the actual event we are waiting will
-    // be fired. The result could be a silent way to create
-    // memory or file descriptor leaks, which is something
-    // we should avoid.
-    if (name !== 'error') {
-      errorListener = (err) => {
-        emitter.removeListener(name, eventListener);
-        reject(err);
+    if (typeof emitter.addEventListener === 'undefined') {
```

Comment:
Perhaps this could be changed to instead check `typeof emitter.addEventListener !== 'function'` and then the check for the opposite later on can be removed since there is already an explicit `return`.

Reply:
@mscdex  good point.
But then I need to change a bit logic around: 
`throw new ERR_INVALID_ARG_TYPE('emitter', 'EventEmitter', emitter);`
This line is at the end of the `once` function.

Maybe to check before the `return new Promise(...)` if the `emitter instanceof EventEmitter` and then throw `ERR_INVALID_ARG_TYPE('emitter', 'EventEmitter', emitter)`?

WDYT?

Reply:
That's probably fine, however I don't know if that would negatively affect `EventTarget` since that is not implemented anywhere in node core.